Backpack-SwiftUI/Tests/SnapshotHelpers/SnapshotHelpers.swift

```diff
@@ -20,23 +20,103 @@ import Foundation
 
-// swiftlint:disable line_length
+/// Create snapshots of a SwiftUI view using several pre-defined options.
+/// This will create as many png snapshots as modes are defined.
+/// - Parameters:
+///       - view: The SwiftUI view to take a snapshot of
+///       - modes: An array of snapshot modes, by default we cover light, dark and right to left
+///       - file: Reference to the calling file
+///       - testName: Reference the calling function
+///       - line: Reference the calling line
 func assertSnapshot<V: View>(
     _ view: V,
-    modes: [UIUserInterfaceStyle] = [.light, .dark],
+    modes: [SnapshotModes] = [.light, .dark, .rtl],
     file: StaticString = #file,
     testName: String = #function,
     line: UInt = #line
 ) {
-    let identifier: (UIUserInterfaceStyle) -> String = { mode in mode == .dark ? "dark-mode" : "light-mode" }
     let view: UIView = UIHostingController(rootView: view).view
     isRecording = false
     modes.forEach { mode in
-        let styleTrait = UITraitCollection(userInterfaceStyle: mode)
-        assertSnapshot(matching: view, as: .image(size: view.intrinsicContentSize, traits: styleTrait), named: identifier(mode), file: file, testName: testName, line: line)
+        assertSnapshot(
+            matching: view,
+            as: .image(size: view.intrinsicContentSize, traits: mode.trait),
+            named: mode.name,
+            file: file,
+            testName: testName,
+            line: line
+        )
     }
+}
+
+/// Test any view across a number of accessibility sizes.
+/// By default we test: default, 135% and 235%
+///
+/// The test view is restricted to 375px in width, to simulate an iPhone SE 3rd gen size
+/// The test view can expand vertically to accomodate as much space as needed (scrollView)
+func assertA11ySnapshot<V: View>(
+    _ view: V,
+    sizes: [ContentSizeCategory] = [.large, .extraExtraExtraLarge, .accessibilityExtraLarge],
+    file: StaticString = #file,
+    testName: String = #function,
+    line: UInt = #line
+) {
+    BPKFont.setDynamicType(enabled: true)
+    let a11yView = VStack(alignment: .leading, spacing: 16) {
+        ForEach(sizes, id: \.self) { size in
+            VStack(alignment: .leading, spacing: 4) {
+                view
+                    .sizeCategory(size)
+                BPKText("\(size)", style: .caption)
+                    .foregroundColor(.textSecondaryColor)
+                    .sizeCategory(.medium)
+            }
+        }
+    }
+    .padding()
+    .frame(maxWidth: 375)
+    
+    assertSnapshot(
+        a11yView,
+        modes: [.custom(named: "a11y", trait: UITraitCollection(userInterfaceStyle: .light))],
+        file: file,
+        testName: testName,
+        line: line
+    )
+}
+
+/// Standardise all snapshot types we allow
+/// For Accessibility, we use .custom(...) as we want to force the view in light mode
+enum SnapshotModes {
+    case light, dark, rtl, custom(named: String, trait: UITraitCollection)
```

Comment:
why not call it `accessibility`? or do you want to open up future possible customisations? (could be both though)